backend/lib/db.js

```diff
@@ -0,0 +1,125 @@
+import assert from 'assert'
+import { ObjectID, MongoClient } from 'mongodb'
+
+import { MongoDBData } from '../../config/projectInfoData.js'
+import { posts_db_name } from '../Utilities/API_utilities'
+class DataBase {
+    constructor () {
+        this.db = {}
+        this.url
+        this.fieldsWanted = {
+            _id: 1,
+            date: 1,
+            type: 1,
+        }
+        this.posts_db_name = posts_db_name
+        if (process.env.NODE_ENV === 'production') {
+            this.url = MongoDBData['mongoData']['productionURL']
+        } else {
+            this.url = MongoDBData['mongoData']['productionURL']
+        }
+        MongoClient.connect(this.url, (err, dbParam) => {
+            assert.equal(null, err)
+            console.log('Successfully connected to MongoDB server. ')
+            this.db = dbParam.db('DisasterResponse')
+            
+            return;
+        })
+    }
+
+    getAll(collection, sortParam = {}) {
+        this.db.collection(collection).find({}).sort(sortParam).toArray((err, result) => {
```

Comment:
@JMStudiosJoe The callback inside `toArray` needs to be removed too, and the return re-added:

Should be something like:
```js
return this.db.collection(collection).find({}).sort(sortParam).toArray()
`

Otherwise it won't return the Promise.